syntax = "proto3";

package protocol;

import "table_data.proto";

message ShardInfo {
  int32 total_shards = 1;
  int32 shard_number = 2;
}

enum Query {
  Q1 = 0;
  Q2 = 1;
  Q3 = 2;
  Q4 = 3;
}

message DataBatch {
  repeated Query query_ids = 1;

  int32 filter_steps = 2;

  repeated ShardInfo shards_info = 3;

  string client_id = 4;

  TableData payload = 5;
}
